### PR TITLE
marshal: improve bytes handling

### PR DESCRIPTION
**Pull request by sobolevn**
Examples:

```python
>>> import marshal
>>> marshal.loads(b'\xe9{\x00\x00\x00')
123
>>> marshal.loads(bytearray(b'\xe9{\x00\x00\x00'))
123
>>> marshal.loads(memoryview(b'\xe9{\x00\x00\x00'))
123
```

### CODE REVIEW

**Review by hauntsaninja — APPROVED** [review]
Lgtm

File: stdlib/marshal.pyi [hunk 1]
```diff
@@ -1,8 +1,9 @@
+from _typeshed import ReadableBuffer
 from typing import IO, Any
 
 version: int
 
 def dump(__value: Any, __file: IO[Any], __version: int = ...) -> None: ...
 def load(__file: IO[Any]) -> Any: ...
```
[the remark this answers is not in the document]
Comment (sobolevn) [reply stdlib/marshal.pyi:7]: I am not double-checking, just commiting 😆

File: stdlib/marshal.pyi [hunk 2]
```diff
@@ -1,5 +1,5 @@
 from _typeshed import ReadableBuffer
```
Comment (JelleZijlstra) [inline stdlib/marshal.pyi]:
```suggestion
from _typeshed import ReadableBuffer, SupportsRead, SupportsWrite
```

File: stdlib/marshal.pyi (hunk 1 above)
[the remark this answers is not in the document]
Comment (JelleZijlstra) [reply stdlib/marshal.pyi]: For the record the relevant code is here https://github.com/python/cpython/blob/main/Python/marshal.c#L1685

File: stdlib/marshal.pyi (hunk 2 above)
Comment (JelleZijlstra) [reply stdlib/marshal.pyi]: Oh you just did that, thanks! GitHub's UI doesn't handle this well: the "Commit" button still shows up on the suggestion but clicking it does nothing. :)

Comment (sobolevn) [reply stdlib/marshal.pyi]: Already done ;)

**Review by JelleZijlstra — APPROVED** [review]

**Pull request merged**